### PR TITLE
Fixes to Android attributable_clients

### PR DESCRIPTION
**Pull request by fbertsch**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [x] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [x] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [x] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v1/query.sql
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v1/query.sql:30]: This is so we have a valid `sample_id` for clients who only reported a search

File: sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v1/query.sql
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v1/query.sql:106]: This may be controversial. I would like the ability to partition on activated users overall. This is still imperfect (first 6 days won't say whether they're activated), but in this way filtering on `activated` will give e.g. DAU for activated users, not the number of activations. For # of activations, we'd look at `activated_on_this_day`

Comment (bani) [reply sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v1/query.sql:106]: Yes, I think `activated_on_this_day` is important to have on the table. I still find it easier to understand the table and write queries when the things we want to count are metrics in a table designed for them to be [additive](https://www.kimballgroup.com/data-warehouse-business-intelligence-resources/kimball-techniques/dimensional-modeling-techniques/additive-semi-additive-non-additive-fact/), and this is a step in that direction if we don't want to actually have it as a number.

I'm not entirely sure we need `activated`. It's not one of our official segments and only once we performed an analysis on how activated users compare to non-activated ones. Do you see that request re-ocurring often enough to justify having a hard to understand table with the two different activated values?

File: sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v1/query.sql
Comment (bani) [inline sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v1/query.sql:58]: I think we should include all clients.

File: sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v1/query.sql
Comment (fbertsch) [reply sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v1/query.sql:106]: I guess I think of `activated` as a dimension, not a fact, especially in regards to this table. Maybe we have two fields:
- `is_activated`: Whether the user is activated, a dimension
- `activations_count`: Number of activations, a fact (and additive)

Looking at `is_actiated` across a number of users was kind of the whole reason I started this PR 😄 where as we're trying to understand how our activated users behave, it makes more sense to target them in certain ways (e.g. marketing)

Comment (bani) [reply sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v1/query.sql:106]: Yes, that would be perfect.

File: sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v1/query.sql
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v1/query.sql:96]: Quick update on what's happening here. The `activations_count` includes clients who were not necessarily active on this day; which means we now need to include a `client_count` to get the actual DAU (but maybe `client_count` is the wrong name -`active_client_count`? `DAU`? Open to suggestions)

File: sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v1/query.sql
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v1/query.sql:118]: Useful now for answering: "Why is this client present on this day?"

**Review by bani — APPROVED** [review]

File: bigquery_etl/cli/query.py
Comment (fbertsch) [inline bigquery_etl/cli/query.py:1182]: @scholtzan this adds the capability to initialize a query file. It first deploys the schema, then runs the query with the templated `is_init` set to `True`.

This feels a little hacked together, and is probably not well tested.

File: bigquery_etl/cli/query.py
Comment (fbertsch) [inline bigquery_etl/cli/query.py:1523]: I had to add this so the `bqetl query schema update` worked for templated queries.

File: bigquery_etl/cli/query.py
Comment (fbertsch) [inline bigquery_etl/cli/query.py:1211]: Everything after this is the same, just indented one level

File: bigquery_etl/util/common.py
Comment (fbertsch) [reply bigquery_etl/util/common.py:67]: I actually like that better, b/c then we don't have to update things in 3 places

**Review by scholtzan — APPROVED** [review]
I'm fine with adding this to unblock this workflow. 
I'm not quite convinced this is the most ideal option. It shouldn't be used for queries where `depends_on_past: true`, for example. Ideally, we'd eventually just have `schema.yaml` files and an easy way trigger backfills when creating these new tables.

File: sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v2/query.sql
Comment (sean-rose) [inline sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v2/query.sql:22]: This Jinja formatting by `bqetl format` is unfortunate.  I just pushed #3952 which should improve such Jinja block formatting once it's merged.

Also, with the new Jinja block formatting, consistently trimming trailing spaces after Jinja block statements with `-%}` seems to produce the most readable SQL when rendered IMO.

File: sql/moz-fx-data-shared-prod/fenix/attributable_clients_v2/view.sql
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/fenix/attributable_clients_v2/view.sql:2]: I thought views should not include version numbers?

Comment (fbertsch) [reply sql/moz-fx-data-shared-prod/fenix/attributable_clients_v2/view.sql:2]: This is a special view; it adds some fields via joins that I need downstream.

On the other hand, we're not quite ready to replace v1, due to some _different_ downstream dependencies. As such, I'm putting the logic in a v2 view, and when we're ready, we'll point the overall view `fenix.attributable_clients` at this view instead.

File: sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v2/query.sql
Comment (fbertsch) [reply sql/moz-fx-data-shared-prod/fenix_derived/attributable_clients_v2/query.sql]: This is great, and looks _much_ better. Thank you @sean-rose!

File: bigquery_etl/cli/query.py
Comment (fbertsch) [reply bigquery_etl/cli/query.py:1198]: Ah missed this. Yes some tests would be good. I'll file a ticket.

**Pull request merged**
